### ISSUE TITLE
Kind Sorter missing list based resources

### ISSUE THREAD

**Issue by jmajoor** [helm/helm#12896]
The kind sorter is missing many of the list based resources.
For example:
Deployment, StatefulSet, ReplicaSet, DaemonSet, Ingress, Service, ConfigMap are defined in the kind sorter, but DeploymentList, StatefulSetList, ReplicaSetList, DaemonSetList, IngressList, ServiceList, ConfigMapList are missing.

There are a few list based resources such as RoleList and SecretList.

Would it be possible for the install/uninstall order to systematically add the corresponding list resource?
Or otherwise make sure that resources are always added in pairs.

**Comment by z4ce** [comment]
Do you have an example of current behavior and desired behavior?

**Comment by jmajoor** [comment]
The current behavior can result in failures due to dependencies across resources, due to missing dependent resources.

For example if you define a ConfigMapList and have a Deployment that refers to config maps created in that list, then the deployment will fail because the ConfigMapList will be created after the deployment instead of before.

Or with a ServiceList and then having an Ingress definition refering to the service. The ServiceList needs to be created before the Ingress.

In other words the current behavior is that all resources not listed in the kind sorter will be created in no particular order at the end, and this can cause failures with dependencies.

Desired behavior is for all List variants of resources to be defined at the same level in the kind sorter as their non-list equivalent, to avoid deployment failures due to cross dependencies.

**Comment by jmajoor** [comment]
Can someone review PR #12915, that addresses this issue?
